_posts/2022-07-19-react-to-full-stack.md

```diff
@@ -0,0 +1,606 @@
+
+    ![](/img/react-to-full-stack/api-14.png)
+
+15. Repeat step 14, for method POST, PUT and DELETE. The exact same Lambda integration is used each time. Remember to always check the 'Use Lambda Proxy Integration' checkbox. After you have finished configuring those methods the /todo resource endpoint should have DELETE, GET, OPTIONS, POST and PUT methods and the /todos resource endpoint should just have GET and OPTIONS methods.
+
+    ![](/img/react-to-full-stack/api-15.png)
+
+16. We now need to setup our authorizer, this will ensure that only authenticated users in our application can make calls to our API and hence interact with our database. Click 'Authorizers' in the side panel.
+
+    ![](/img/react-to-full-stack/api-16.png)
+
+17. Click 'Create New Authorizer'.
+
+    ![](/img/react-to-full-stack/api-17.png)
+
+18. Give your authorizer a name, select 'Cognito' as the 'Type', ensure you have selected the same region that you provisioned your cognito user pool in earlier, then click the box and select the user pool you created earlier. Enter 'Authorization' into the 'Token Source' and leave 'Token Validation' empty. Click 'Create'.
+
+    ![](/img/react-to-full-stack/api-18.png)
+
+19. Using the side panel, navigate back to 'Resources'.
+
+    ![](/img/react-to-full-stack/api-19.png)
+
+20. Click the 'DELETE' method, then click 'Method Request'.
+
+    ![](/img/react-to-full-stack/api-20.png)
+
+21. Click the pencil icon next to 'Authorization'.
+
+    ![](/img/react-to-full-stack/api-21.png)
+
+22. Click the box and a drop down list will appear, select the cognito authorizer we previously created, then click the check icon to confirm.
+
+    ![](/img/react-to-full-stack/api-22.png)
+
+23. Repeat steps 20 to 22 for methods: GET, POST, and PUT under resource endpoint '/todo', then the GET method under resource endpoint '/todos' (DO NOT ATTACH AN AUTHORIZER TO THE OPTIONS METHODS). Once the relevant method requests have the authorizers attached click 'Actions' => 'Deploy API'.
+
+    ![](/img/react-to-full-stack/api-23.png)
+
+24. You will then see this window. For 'Deployment stage' select '[New Stage]', give your stage a name, then click 'Deploy'.
+
+    ![](/img/react-to-full-stack/api-24.png)
+
+25. You will then be brought to this page. Copy your "invoke URL".
+
+    ![](/img/react-to-full-stack/api-25.png)
+
+    Now open constants.ts (todo-app > src > constants > constants.ts). Paste your invoke URL in both the TODOS_API and TODO_API values. You will also need to add the relevant endpoints to each API constant. So add '/todos' to the end of the invoke URL in TODOS_API and add '/todo' to the end of your invoke URL for TODO_API. At this point your API variables should look like this.
+
+    ```javascript
+    export const TODOS_API =
+      "https://jgb9txcg6c.execute-api.ap-southeast-2.amazonaws.com/prod/todos";
+    export const TODO_API =
+      "https://jgb9txcg6c.execute-api.ap-southeast-2.amazonaws.com/prod/todo";
```

Comment:
likewise with providing these endpoints, security risk? maybe

Reply:
I considered this and then deleted the resources in my AWS account, so I think it should be all good.